include/WiFiCredentials.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <nonstd/span.hpp>
+
+#include <esp_wifi_types.h>
+
+#include <array>
+#include <vector>
+#include <cstdint>
+
+namespace fs { class File; }
+class String;
+
+namespace OpenShock {
+  class WiFiCredentials {
+    WiFiCredentials() = default;
+  public:
+    static bool Load(std::vector<WiFiCredentials>& credentials);
+
+    WiFiCredentials(std::uint8_t id, const wifi_ap_record_t* record);
+    WiFiCredentials(std::uint8_t id, const String& ssid, const String& password);
+
+    constexpr std::uint8_t id() const noexcept {
+      return _id;
+    }
+
+    constexpr nonstd::span<const char> ssid() const noexcept {
+      return nonstd::span<const char>(reinterpret_cast<const char*>(_ssid), _ssidLength);
+    }
+    constexpr nonstd::span<const std::uint8_t> ssidBytes() const noexcept {
+      return nonstd::span<const std::uint8_t>(_ssid, _ssidLength);
+    }
```

Comment:
So if my understanding is correct, this is just some zero-copy type casting for the same set of bytes?